public/src/MockGRPCManager.ts

```diff
@@ -456,6 +456,14 @@ export class MockGrpcManager {
         return this.grpcSend<Submissions>(submissions)
     }
 
+    public getSubmission(courseID: number, submissionID: number): Promise<IGrpcResponse<Submission>> {
+        if (!this.enrollments.getEnrollmentsList().find(e => e.getCourseid() === courseID && e.getUserid() === this.currentUser?.getId())) {
```

Comment:
I think I would prefer to split this into two lines for clarity. One to find the enrollment `e`, and another for the check that it exists or not.